### DISCUSSION TITLE
B2 ERROR with "make PythonAPI" and "make launch"

### DISCUSSION THREAD

**Discussion by ThibaultFy** [carla-simulator/carla#4534] [Q&A]
I'm having trouble with my Windows build for 0.9.12. I'm getting the following message:

> -[install_boost]: [B2 ERROR] An error ocurred while installing using "b2.exe". 
> -[install_boost]: Exiting with error...

Thanks for your help !

**Comment by ThibaultFy** [comment]
My problem solved itself by removing the build folder and changing the path on my repo... I don't have more explanation than this...

**Reply by llllllyan92** [reply]
Hi,
I ran into the same issue and moved everything from the repo path **D:\Carla\CL\carla** to **D:\Carla\CL** then it worked.
I'm guessing the original path was still too long to be handled.
Thanks a lot for the hint!

**Reply by llllllyan92** [reply]
An update:
I removed the repo and redid the whole process again, and found it wasn't, in my case, due to the path.

I observed linking errors for targets failed such as:
**link: extra operand** '/out:build\\boost\\bin.v2\\libs\\python\\build\\msvc-14.2\\release\\link-static\\python-3.8\\threading-multi\\libboost_python38-vc142-mt-x64-1_72.lib'

but after I ran link command manually for these targets (I happened to do the same ystd before moving my repo) and retry make PythonAPI, this b2 error was gone:
D:\Carla\cl\Build\boost-1.72.0-source> link /lib /NOLOGO  /out:"build\boost\bin.v2\libs\python\build\msvc-14.2\release\link-static\python-3.8\threading-multi\libboost_python38-vc142-mt-x64-1_72.lib" @"build\boost\bin.v2\libs\python\build\msvc-14.2\release\link-static\python-3.8\threading-multi\libboost_python38-vc142-mt-x64-1_72.lib.rsp"

**Reply by dhruv-g334** [reply]
I am still getting the issue. It seems like I don't have the "libboost_python38-vc142-mt-x64-1_72.lib.rsp" file on the desired location. What should I do?

**Reply by LBroKerR** [reply]
I have this problem too
